smg-automotive/components-pkg
# fix: added custom component

References https://autoricardo.atlassian.net/browse/DM-605

## Motivation and context

 Improve accessibility, security and SEO while using a button as link.

## How to test
**On components library:** please check the `Button as Link` as variation on the button and check the DOM on console to verify the link tag. 
YOU can also play around with the next cases on the `DevOverlay` component where we import the button.
_Cases where type triggers an error:_
```
<Button as="a">
<Button href="goo">
<Button isExternal={true}>
<Button rel="d">
<Button onClick={hideDevOverlay} href="asdasdsa" isExternal={true}>
<Button href="asdasdsa" isExternal={true}>
<Button as="a" isDisabled={true}>
```

## Thank you 🦭
